<?xml version="1.0" ?>

<launch>
    <!-- obstalce visualize -->
    <include file="$(find cbf_ros)/launch/obstacle_test.launch"/>
    <include file="$(find cbf_controllers)/test/dcbf_robot.launch"/>

    <!-- visualization config -->
    <arg name="rviz"               default="true" />
    <arg name="description_name"   default="legged_robot_description"/>
    <arg name="multiplot"          default="false"/>

    <!-- The task file for the mpc. -->
    <arg name="taskFile" default="$(find cbf_controllers)/config/task.info"/>
    <!-- The reference related config file of the robot -->
    <arg name="referenceFile" default="$(find cbf_controllers)/config/reference.info"/>
    <!-- The URDF model of the robot -->
    <arg name="urdfFile" default="$(find legged_unitree_description)/urdf/a1.urdf"/>
    <!-- The file defining gait definition -->
    <!-- <arg name="gaitCommandFile" default="$(find legged_controllers)/config/go1/gait.info"/> -->
    <arg name="gaitCommandFile"   default="$(find ocs2_legged_robot)/config/command/gait.info"/>

    <group if="$(arg rviz)">
        <param name="$(arg description_name)" textfile="$(arg urdfFile)"/>
        <arg name="rvizconfig" default="$(find cbf_ros)/rviz/rviz.rviz"/>
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)" output="screen"/>
    </group>
    <!-- make the files into global parameters -->
    <param name="multiplot"         value="$(arg multiplot)"/>
    <param name="taskFile"          value="$(arg taskFile)" />
    <param name="referenceFile"     value="$(arg referenceFile)" />
    <param name="urdfFile"          value="$(arg urdfFile)" />
    <param name="gaitCommandFile"   value="$(arg gaitCommandFile)"/>

    <node pkg="cbf_ros" type="cbf_sqp_mpc" name="cbf_sqp_mpc"
          output="screen" launch-prefix=""/>

    <node pkg="cbf_ros" type="cbf_dummy" name="cbf_dummy"
          output="screen" launch-prefix="gnome-terminal --"/>
    
    <node pkg="cbf_ros" type="cbf_target" name="cbf_target" output="screen"/>
    

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen" launch-prefix="gnome-terminal --"/>
</launch>
